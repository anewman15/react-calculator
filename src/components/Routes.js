/*
 eslint-disable no-unused-vars
*/

import { Switch, Route } from 'react-router-dom';
import Home from './Home';
import Calculator from './Calculator';
import QuoteOfTheDay from './QuoteOfTheDay';

const Routes = () => (
  <Switch>
    <Route exact path='/' component={Home} />
    <Route path='/calculator' component={Calculator} />
    <Route exact path='/quote-of-the-day' component={QuoteOfTheDay} />
  </Switch>
);

export default Routes;
