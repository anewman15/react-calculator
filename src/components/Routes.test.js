/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import Routes from './Routes';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Routes', () => {
  it('should render a navigation bar and routes of the site', () => {
    const tree = renderer
      .create(<Routes />)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});