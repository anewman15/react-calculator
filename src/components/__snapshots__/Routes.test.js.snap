// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render a navigation bar and routes of the site 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light bg-success"
    >
      <a
        className="text-white font-weight-bolder navbar-brand"
        href="/"
      >
        Over the Counter Calculator
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="ml-auto text-white navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="text-white nav-link"
            data-rb-event-key="/"
            href="/"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Home
          </a>
          <a
            className="text-white nav-link"
            data-rb-event-key="/calculator"
            href="/calculator"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Calculator
          </a>
          <a
            className="text-white nav-link"
            data-rb-event-key="/quote-of-the-day"
            href="/quote-of-the-day"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Quote of the Day
          </a>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="container container-fluid"
  >
    <h1
      className="text-center mt-4"
    >
      Over the Counter Calculator
    </h1>
    <p
      className="display-5 my-4"
    >
      This is one of the primitive calculators. Built with React. I'll probably improve it over time. Can be used for basic calculations using arithmatic oeprations:
    </p>
    <div
      className="list-group w-25 mx-auto list-group"
      onKeyDown={[Function]}
    >
      <div
        className="list-group-item list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Addition
      </div>
      <div
        className="list-group-item list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Subtraction
      </div>
      <div
        className="list-group-item list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Multiplication
      </div>
      <div
        className="list-group-item list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Division
      </div>
      <div
        className="list-group-item list-group-item"
        data-rb-event-key={null}
        disabled={false}
        onClick={[Function]}
      >
        Percent Calculation
      </div>
    </div>
    <div
      className="text-center my-5"
    >
      <a
        className="text-white w-50 btn btn-success"
        href="/calculator"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Try It Now
      </a>
    </div>
  </div>,
]
`;
